rfc/0012-policy-report.md

```diff
@@ -95,12 +96,429 @@ results from Kubernetes policy engines (like Kubewarden or Kyverno) but can
+users of the cluster should only have READ access to them. They should not be
+able to create, delete or change their contents. This can be done with Kubernetes
+RBAC rules.
+
+### `PolicyReport` detailed overview
+
+For each Namespace, Kubewarden will find all the policies that are interested
+about actions happening inside of it. This is going to be a mix of
+some `ClusterAdmissionPolicy` policies and all the `AdmissionPolicy` policies
+that are defined inside of the Namespace.
+
+Next, this list of policies is going to be filtered: only policies interested
+in `CREATE` events are going to be considered.
+The audit report is not going to simulate `DELETE` and `UPDATE` events.
+
+For each relevant policy, Kubewarden will find all the Kubernetes resources that
+are 1) defined inside of the Namespace and 2) are relevant to the policy. Kubewarden
+will then simulate the `CREATE` event of this resource and keep track of the policy
+evaluation result.
+
+All the results are going to be collected into the `PolicyReport` object
+defined inside of the Namespace. There's going to be only one `PolicyReport`
+per audited resource.
+
+The `PolicyReport` resource has the following fields:
+
+- `apiVersion` and `kind`: these are defined by the CRDs
+- `metadata`: this is the usual `meta/v1.ObjectMeta` resource. In the
+  beginning we will set only the `name` field. We will do that by using the
+  audited resource UID
+- `scope`: this is a `core/v1.ObjectReference` resource. We will use that to
+  reference the audited resource that has been analyzed
+- `scopeSelector`: we will not use that, because we are using the `scope` field
+- `summary`: this is an object defined by a CRD `PolicyReportSummary`. It's made
+  of the following fields:
+  - `pass`: count of policies whose requirements were met
+  - `fail`: count of policies whose requirements were not met
+  - `warn:`: we will not set it, it doesn't relate to us
+  - `error`: count of policies that could not be evaluated.
+    The policies could not be evaluated because of errors
+    (for example: a runtime panic of a broken policy)
+  - `skip`: count of policies that were not selected for evaluation.
+    We will allow some policies to be excluded from the background checks, more
+    on that later.
+- `results` this probably the most important field. It's a list of `PolicyReportResult`
+  objects
+
+The structure of the `PolicyReportResult` is flexible. This is how we are going
+to use its fields:
+
+- `source`: identifier for the policy engine that manages this report. We will
+  set that to be `kubewarden`
+- `policy`: this is the name of the policy that evaluated the resource. This
+  could be both a `ClusterAdmissionPolicy` or a `AdmissionPolicy`.
+  Because of that we cannot use the name of the policy, there could be clashes
+  between cluster and namespaced policy names. We will instead use the following
+  convention to build the contents of this field: `<[cap|ap]>-<policy name>`.
```

Comment:
in the new implementation we use the policy unique name instead (e.g. clusterwide-policyname or namespaced-default-policy)